cfgov/unprocessed/apps/analytics-gtm/js/bah-explainers.js

```diff
@@ -0,0 +1,138 @@
+  addEventListenerToSelector,
+  closest,
+  track
+} from './util/analytics-util';
+
+/*
+Owning a Home /owning-a-home/loan-estimate/
+Owning a Home /owning-a-home/closing-disclosure/
+*/
+
+export default function( label ) {
+
+  const trackingLabel = `${ label } Interaction`;
+
+  const expandableStates = {};
+  let lastExpandable;
+
+  /**
+   * Record state (expanded or collapsed) of expandables.
+   * @param {string} id - The unique HTML ID of an expandable.
+   */
+  function recordExpandableState( id ) {
+    if ( !expandableStates[id] ) {
+      expandableStates[id] = true;
+    } else if ( lastExpandable === id ) {
+      expandableStates[id] = !expandableStates[id];
+    }
+    lastExpandable = id;
+  }
+
+  /**
+   * Record state (expanded or collapsed) of expandables.
+   * @param {string} expandable - An expandable HTML element.
+   * @returns {boolean} True if the expandable is animating, false otherwise.
+   */
+  function isAnimatingExpandable( expandable ) {
+    let isAnimating = false;
+
+    if ( expandable.classList.contains( 'o-expandable__expanding' ) ||
+         expandable.classList.contains( 'o-expandable__collapsing' ) ) {
+      isAnimating = true;
+    }
+
+    return isAnimating;
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackTabLinkClick( event ) {
+    const target = event.target;
+    const text = target.textContent.trim();
+    track( trackingLabel, 'Tab click', text );
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackFormExplainerPageLinkClick( event ) {
+    const target = event.target;
+    const pageNumber = 'Page ' + target.getAttribute( 'data-page' );
+    track( trackingLabel, 'Page link click', pageNumber );
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackFormExplainerPageButtonClick( event ) {
+    const target = event.currentTarget;
+    const currentPageDom =
+      document.querySelector( '.form-explainer_page-link.current-page' );
+    const currentPage = 'Page ' + currentPageDom.getAttribute( 'data-page' );
+    let action = 'Next Page button clicked';
+    if ( target.classList.contains( 'prev' ) ) {
+      action = 'Previous Page button clicked';
+    }
+    track( trackingLabel, action, currentPage );
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackExpandableTargetsClick( event ) {
+    const elem = event.currentTarget;
+    const expandable = elem.parentNode;
+    const expandableID = expandable.id;
+    if ( isAnimatingExpandable( expandable ) ) { return; }
+    recordExpandableState( expandableID );
+
+    const tab = closest( elem, '.explain' ).querySelector( '.active-tab' );
+    const tabText = tab.querySelector( '.tab-label' ).textContent.trim();
+    let action = 'Expandable collapsed - ' + tabText;
+    const label = elem.querySelector( '.o-expandable_label' );
+    let text = '';
```

Comment:
Why `let` and define on the next line? Doesn't look like this gets redefined at all.

Reply:
hm, yeah not sure what happened there. Fixed!